File: src/main/resources/scala-native/sfml/network/TcpListener.c
#include <SFML/Network.h>

sfTcpSocket* sfTcpListener_accept_fix(sfTcpListener* listener)
{
    sfTcpSocket* connected;

    if (sfTcpListener_accept(listener, &connected) != sfSocketDone) {
        return NULL;
    }
    return connected;
}

sfSocketStatus sfTcpListener_listen_fix(sfTcpListener* listener, unsigned short port, sfIpAddress* address) {
    return sfTcpListener_listen(listener, port, *address);
}

File: src/main/scala/sfml/internal/network/TcpListener.scala
package sfml
package internal
package network

import scalanative.unsafe.*

@link("sfml-network")
@extern object TcpListener:
    type sfTcpListener

    def sfTcpListener_create(): Ptr[sfTcpListener] = extern
    def sfTcpListener_destroy(listener: Ptr[sfTcpListener]): Unit = extern

    def sfTcpListener_getLocalPort(listener: Ptr[sfTcpListener]): CUnsignedShort = extern

    def sfTcpListener_accept_fix(listener: Ptr[sfTcpListener]): Ptr[TcpSocket.sfTcpSocket] = extern

    def sfTcpListener_listen_fix(
        listener: Ptr[sfTcpListener],
        port: CUnsignedShort,
        address: Ptr[IpAddress.sfIpAddress]
    ): SocketStatus.sfSocketStatus = extern
